src/tools/ApiChecker/IdentifyMissingTests.cs

```diff
@@ -0,0 +1,164 @@
+                type => type.GetMethods().SelectMany(
+                    method => method.GetCustomAttributes<PlaywrightSharp.Tests.PlaywrightTestAttribute>()));
+
+            int potentialMatches = 0;
+            int fullMatches = 0;
+            int noMatches = 0;
+            int totalTests = 0;
+
+            List<(string, string)> missingTests = new();
+            List<KeyValuePair<(string FileName, string TestName), List<(string, string)>>> invalidMaps = new();
+            foreach (var atx in attributes)
+            {
+                totalTests++;
+
+                // a test can either be a full match, a partial (i.e. just the test name) or no match
+                var potentialMatch = _testPairs.Where(x => string.Equals(x.TestName, atx.TestName, StringComparison.InvariantCultureIgnoreCase));
+                if (!potentialMatch.Any())
+                {
+                    noMatches++;
+                    missingTests.Add((atx.FileName, atx.TestName));
+                }
+                else if (potentialMatch.Count(x => string.Equals(x.FileName, atx.TrimmedName, StringComparison.InvariantCultureIgnoreCase)) == 1)
+                {
+                    fullMatches++;
+                    continue;
+                }
+                else
+                {
+
+                    invalidMaps.Add(new KeyValuePair<(string FileName, string TestName), List<(string, string)>>((atx.TrimmedName, atx.TestName), potentialMatch.ToList()));
+                    potentialMatches++;
+                }
+            }
+
+            Console.ForegroundColor = ConsoleColor.Green;
+            Console.WriteLine($"Total matching tests: {fullMatches}/{totalTests}.");
+            Console.ResetColor();
+
+            Console.ForegroundColor = ConsoleColor.Yellow;
+            Console.WriteLine($"Total tests found by name, but not by file: {potentialMatches}/{totalTests}.");
+            Console.ResetColor();
+
+            foreach (var invalidTest in invalidMaps)
+            {
+                Console.WriteLine($"{invalidTest.Key.FileName}: {invalidTest.Key.TestName}");
+                foreach (var items in invalidTest.Value)
+                {
+                    Console.WriteLine($"\t{items.Item1}: {items.Item2}");
+                }
+            }
+
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($"Total missing tests: {noMatches}/{totalTests}.");
+            Console.ResetColor();
+
+            foreach (var invalidTest in missingTests)
+            {
+                Console.WriteLine($"{invalidTest.Item1}: {invalidTest.Item2}");
+            }
+
+            Console.WriteLine($"Found/Mismatched/Missing: {fullMatches}/{potentialMatches}/{noMatches} out of {totalTests}");
+        }
+
+        private static void MapTestsCases(DirectoryInfo directoryInfo, IdentifyMissingTestsOptions options)
+        {
+            // get the sub-directories
+            if (options.Recursive)
```

Comment:
When wouldn't we want to go recursive?

Reply:
I could see a scenario of having tests for new features in a subfolder with subfolders. I figured it was "cheap" enough to implement.